src/tla/Apalache.tla

```diff
@@ -53,13 +53,30 @@ SetAsFun(S) ==
     IN
     [ x \in Dom |-> CHOOSE y \in Rng: <<x, y>> \in S ]
 
+(**
+ * A sequence constructor that is avoiding a function constructor.
+ * Since Apalache is typed, this operator is more efficient than
+ * FunAsSeq([ i \in 1..N |-> F(i) ]). Apalache requires N to be
+ * a constant expression.
+ *
+ * @type: (Int, (Int -> a)) => Seq(a);
+ *)
+LOCAL INSTANCE Integers
+MkSeq(N, F(_)) ==
+    \* This is the TLC implementation. Apalache does it differently.
+    [ i \in (1..N) |-> F(i) ]
+
 (**
  * As TLA+ is untyped, one can use function- and sequence-specific operators
  * interchangeably. However, to maintain correctness w.r.t. our type-system,
  * an explicit cast is needed when using functions as sequences.
+ *
+ * @type: ((Int -> a), Int) => Seq(a);
```

Comment:
Not sure about Sany, will this annotation stick to `LOCAL INSTANCE` or `FunAsSeq ==...` ?

Reply:
Actually, we can remove this instance now, as it was needed by default implementation of `FunAsSeq`